test/framework/easyconfig.py

```diff
@@ -1204,6 +1203,49 @@ def test_dump_extra(self):
 
         dumped_ec = EasyConfig(testec)
 
+    def test_dump_template(self):
+        """ Test EasyConfig's dump() method for files containing templates"""
+        rawtxt = '\n'.join([
+            'easyblock = "EB_toy"',
+            '',
+            'name = "Foo"',
+            'version = "0.0.1"',
+            '',
+            'homepage = "http://foo.com/"',
+            'description = "foo description"',
+            '',
+            "toolchain = {'version': 'dummy', 'name': 'dummy'}",
+            '',
+            "sources = ['foo-0.0.1.tar.gz']",
+            '',
+            'preconfigopts = "--opt1=%s" % name',
+            'configopts = "--opt2=0.0.1"',
+            '',
+            "sanity_check_paths = {'files': ['files/foo/bar'], 'dirs':[] }",
```

Comment:
change to `'files/foo/foobar'`, which should result in `'files/%(namelower)s/foobar'`, i.e. the `foo` in `'foobar'` does not get replaced with the matching template?